crates/topos-tce-api/src/runtime/sync_task.rs

```diff
@@ -0,0 +1,179 @@
+                        if let Entry::Vacant(entry) = source.entry(source_subnet_id) {
+                            entry.insert(TargetStreamPosition {
+                                target_subnet_id: *target_subnet_id,
+                                source_subnet_id,
+                                position: 0,
+                                certificate_id: None,
+                            });
+                        }
+                    }
+                }
+
+                for TargetStreamPosition {
+                    target_subnet_id,
+                    source_subnet_id,
+                    position,
+                    ..
+                } in source.values_mut()
+                {
+                    if self.cancel_token.is_cancelled() {
+                        self.status = SyncTaskStatus::Cancelled;
+                        return SyncTaskStatus::Cancelled;
+                    }
+                    if let Ok(certificates_with_positions) = self
+                        .storage
+                        .fetch_certificates(FetchCertificatesFilter::Target {
+                            target_stream_position: CertificateTargetStreamPosition {
+                                target_subnet_id: *target_subnet_id,
+                                source_subnet_id: *source_subnet_id,
+                                position: (*position).into(),
+                            },
+                            limit: 100,
+                        })
+                        .await
+                    {
+                        collector.extend(certificates_with_positions)
+                    }
+                }
+            }
+
+            for (CertificateDelivered { certificate, .. }, position) in collector {
+                info!(
+                    "Stream sync task for {} is sending {}",
+                    self.stream_id, certificate.id
+                );
+
+                if let FetchCertificatesPosition::Target(CertificateTargetStreamPosition {
+                    target_subnet_id,
+                    source_subnet_id,
+                    position,
+                }) = position
+                {
+                    if let Err(e) = self
+                        .notifier
+                        .send(StreamCommand::PushCertificate {
+                            positions: vec![TargetStreamPosition {
+                                target_subnet_id,
+                                source_subnet_id,
+                                position: *position,
+                                certificate_id: Some(certificate.id),
+                            }],
+                            certificate,
+                        })
+                        .await
+                    {
+                        error!("Error sending certificate to stream: {}", e);
+                        self.status = SyncTaskStatus::Error;
+                        return SyncTaskStatus::Error;
```

Comment:
Can't we let `SyncTaskStatus::Error` carry the error so that consumers can use it directly?